aiidalab_widgets_base/data/__init__.py

```diff
@@ -34,7 +34,10 @@ def __init__(self, value=0, description="Select functional group", **kwargs):
         self.style = {"description_width": "initial"}
         self.layout = {"width": "initial"}
         super().__init__(
-            value=value, description=description, options=FUNCTIONAL_GROUPS, **kwargs
+            value=value,
+            description=description,
+            options=[(key, value) for key, value in FUNCTIONAL_GROUPS.items()],
```

Comment:
The options is a tuple and using mutable as default argument is non-pythonic (see https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html#anti-pattern).

Reply:
> Also this is not a default argument, but parameter passed to the constructor.

Sorry, you are right, this is not the default argument.

> Options should be a list of tuples, no?

I think it can be an iterable type, a list is fine. But when ipywidget internally will treat it to a tuple with the function below. I encountered the case that I compared the empty options with a list in the condition and the test failed which confused me at first time.

https://github.com/jupyter-widgets/ipywidgets/blob/52663ac472c38ba12575dfb4979fa2d250e79bc3/python/ipywidgets/ipywidgets/widgets/widget_selection.py#L107

Reply:
Yeah, this is a gotcha. I agree using tuples where possible is nice to indicate immutability. I've done this in all but one place where I could not easily use a tuple instead of a list.

Reply:
> I've done this in all but one place where I could not easily use a tuple instead of a list.

I think that's fine.